ouroboros-consensus-cardano/src/shelley/Ouroboros/Consensus/Shelley/Ledger/Ledger.hs

```diff
@@ -320,6 +321,32 @@ instance ShelleyCompatible proto era => UpdateLedger (ShelleyBlock proto era)
 type instance TxIn (LedgerState (ShelleyBlock proto era)) = SL.TxIn
 type instance TxOut (LedgerState (ShelleyBlock proto era)) = Core.TxOut era
 
+-- | We use this newtype only to alter MemPack serialization. LSM trees use an
+-- index that looks at the first 8 bytes so it is important to put the index
+-- first such that we avoid all TxIns from the same tx to imply a collision in
+-- the LSM index.
+newtype LSMTxIn = LSMTxIn {lsmTxIn :: SL.TxIn}
+
+instance MemPack LSMTxIn where
+  packedByteCount = packedByteCount . lsmTxIn
+  packM (LSMTxIn (SL.TxIn txid txix)) = packM txix >> packM txid
+  unpackM = do
```

Comment:
We might want to consider avoiding the `do` notation, especially to be consistent with the definition above.